addons/udes_stock/tests/test_stock_location_picking_zone.py

```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
+
+from odoo.addons.udes_stock.tests.common import BaseUDES
+
+
+class PickingZoneTestCase(BaseUDES):
+    def create_location(self, name, **kwargs):
+        Location = self.env["stock.location"]
+
+        vals = {"name": name, "location_id": self.test_location_01.id}
+        vals.update(kwargs)
+        return Location.create(vals)
+
+    def test_get_picking_zone_self_is_zone(self):
+        """
+        Test that a location that is a picking zone returns itself when
+        get_picking_zone() is called.
+        """
+        loc = self.create_location("picking_zone_test_location", u_is_picking_zone=True)
+        self.assertEqual(loc.get_picking_zone(), loc)
+
+    def test_get_picking_zone_a_parent_is_zone(self):
+        """
+        Test that when a parent location is a picking zone it is returned
+        correctly.
+        """
+        loc = self.create_location("picking_zone_test_location")
+        self.test_location_01.u_is_picking_zone = True
+        self.assertEqual(loc.get_picking_zone(), self.test_location_01)
+
+    def test_get_picking_zone_no_zone(self):
+        """
+        Test that when no parent location is a picking zone, an empty recordset
+        is returned.
+        """
+        Location = self.env["stock.location"]
+
+        loc = self.create_location("picking_zone_test_location")
+        self.assertEqual(loc.get_picking_zone(), Location.browse())
```

Comment:
Nit (feel free to ignore/resolve): Could just do `self.assertFalse`, would remove the need for `Location` and `Location.browse()`. But I suppose how it is now is more explicit.

Reply:
Good point, changed.